framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -266,8 +353,12 @@ private static boolean isIntLiteral(String s, FlowExpressionContext context) {
         if (context.parsingMember) {
             return false;
         }
-        Matcher intMatcher = INT_PATTERN.matcher(s);
-        return intMatcher.matches();
+        try {
+            Expression e = parseExpression(s);
```

Comment:
Don't evaluate `parseExpression` twice.  It is inefficient and it will confuse readers as to why you need to do it twice.

You can write:

```
Expression e;
try {
  e = parseExpression(s);
} catch (ParseProblemException pe) {
  return false;
}
return e.getClass().equals(IntegerLiteralExpr.class);
```

but in this case I think it's even better to write 

```
try {
  return parseExpression(s).getClass().equals(IntegerLiteralExpr.class);
} catch (ParseProblemException pe) {
  return false;
}
```

because it is obvious that only `parseExpression` can throw `ParseProblemException`.

This is now short and clear.  You can make a similar change elsewhere.